src/main/webapp/app/core/util/alert.service.ts

```diff
@@ -129,6 +129,18 @@ export class AlertService {
                             break;
                         }
                         this.addErrorAlert(httpErrorResponse.error.title, httpErrorResponse.error.message, httpErrorResponse.error.params);
+                    } else if (this.isErrorFromStatusCode(httpErrorResponse.status)) {
+                        let translationKey = httpErrorResponse.headers.get('x-artemisapp-error') ?? undefined;
+                        let message = httpErrorResponse.headers.get('x-artemisapp-message');
```

Comment:
Probably not that important, but this was the only thing that caught my eye - it looks a little inconsistent (`string | null` vs. `string | undefined`, I assume?). Maybe you could define a method that extracts a header from a http response and returns a `string | undefined`.

Reply:
As you do not seem to have a strong opinion on this I would leave it as it is for now. 

In case changes are required I would directly address setting the null values in `addErrorAlert` to undefined, as this is the reason why `undefined` is set here as fallback value.